pkg/controllers/machine/garbagecollect.go

```diff
@@ -32,20 +36,28 @@ type GarbageCollect struct {
 	lastChecked   *cache.Cache
 }
 
-func (e *GarbageCollect) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
+func (g *GarbageCollect) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reconcile.Result, error) {
 	if !machine.StatusConditions().GetCondition(v1alpha5.MachineCreated).IsTrue() {
 		return reconcile.Result{}, nil
 	}
 	// If there is no node representation for the machine, then check if there is a representation at the cloudprovider
-	if _, err := nodeForMachine(ctx, e.kubeClient, machine); err == nil || !IsNodeNotFoundError(err) {
+	if _, err := machineutil.NodeForMachine(ctx, g.kubeClient, machine); err == nil || !machineutil.IsNodeNotFoundError(err) {
 		return reconcile.Result{}, nil
 	}
-	if _, expireTime, ok := e.lastChecked.GetWithExpiration(client.ObjectKeyFromObject(machine).String()); ok {
+	if _, expireTime, ok := g.lastChecked.GetWithExpiration(client.ObjectKeyFromObject(machine).String()); ok {
 		return reconcile.Result{RequeueAfter: time.Until(expireTime)}, nil
 	}
-	if _, err := e.cloudProvider.Get(ctx, machine.Status.ProviderID); cloudprovider.IsMachineNotFoundError(err) {
-		return reconcile.Result{}, client.IgnoreNotFound(e.kubeClient.Delete(ctx, machine))
+	if _, err := g.cloudProvider.Get(ctx, machine.Status.ProviderID); cloudprovider.IsMachineNotFoundError(err) {
+		if err = g.kubeClient.Delete(ctx, machine); err != nil {
+			return reconcile.Result{}, client.IgnoreNotFound(err)
```

Comment:
No, I admit that the two concepts are a tad confusing, but there is a concept of a "Cloudprovider Machine" and the K8s CRD "Machine". If we don't find the "cloudprovider machine" (you can also think of the instance), then we should remove the CR "Machine" and ignore an error if the CR is already gone.

Reply:
Ah, would it be better to call it UnderlyingMachine, or RepresentedMachine, or BackingMachine? I can definitely see how this would be confusing to any devs in the future. If not changing the name I'd recommend a comment here.